pkg/auto/sccexporter/mqtt.go

```diff
@@ -0,0 +1,113 @@
+package sccexporter
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"go.uber.org/zap"
+
+	"github.com/smart-core-os/sc-bos/pkg/auto/sccexporter/config"
+	"github.com/smart-core-os/sc-golang/pkg/trait"
+)
+
+type Device struct {
+	Name string `json:"name"`
+	// Data is a map of device data, the key is the trait name and the value is JSON data. Could be data or metadata depending on the message type
+	Data map[trait.Name]string `json:"data,omitempty"`
+}
+
+type message struct {
+	Agent     string    `json:"agent"`
+	Device    Device    `json:"device"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+type sccConnector struct {
+	logger     *zap.Logger
+	messagesCh chan message
+	mqttCfg    config.Mqtt
+	mqttClient mqtt.Client
+}
+
+func newSccConnector(logger *zap.Logger, mqttCfg config.Mqtt, client mqtt.Client) *sccConnector {
+	s := &sccConnector{
+		logger:     logger,
+		messagesCh: make(chan message, 100),
+		mqttCfg:    mqttCfg,
+		mqttClient: client,
+	}
+	return s
+}
+
+// publishToScc listens on the messages channel and publishes messages to the SCC MQTT broker
+// do not call this more than once per sccConnector instance
+func (s *sccConnector) publishToScc(ctx context.Context) error {
+
+	token := s.mqttClient.Connect()
+	if !token.WaitTimeout(s.mqttCfg.ConnectTimeout.Duration) {
+		return fmt.Errorf("timeout connecting to mqtt broker")
+	}
```

Comment:
Should there be a retry here? If we can't connect to MQTT the auto will stop without anybody noticing unless you read the logs.